src/gridstack.ts

```diff
@@ -2030,16 +2030,35 @@ export class GridStack {
     // vars shared across all methods
     let cellHeight: number, cellWidth: number;
 
+    // creates a reference element for tracking the right position after scaling
+    const testEl = document.createElement('div');
+    Utils.addElStyles(testEl, {
+      opacity: '0',
+      position: 'fixed',
+      top: 0 + 'px',
+      left: 0 + 'px',
+      width: '1px',
+      height: '1px',
+      zIndex: '-999999',
+    });
+
     let onDrag = (event: DragEvent, el: GridItemHTMLElement, helper: GridItemHTMLElement) => {
       let node = el.gridstackNode;
       if (!node) return;
 
       helper = helper || el;
+      helper.appendChild(testEl);
+      const testElPosition = testEl.getBoundingClientRect();
+      helper.removeChild(testEl);
+      const dragScale = {

```

Comment:
you will need the offset as well, which makes now 3 copies of that code so it should be extracted into a util method instead.

Reply:
I moved some redundant logic for that to utils. I also now made it possible for elements dragged from outside to be scaled down when entering the grid, and scaled back up when they leave. Also their position were following the unscaled grid so I fixed that too. 
I am not sure which offset you mean but I did adjust the offset for the outside item's helpers when they enter the grid.

https://github.com/gridstack/gridstack.js/assets/23745606/43abe1e9-71a8-469e-915e-cfcb4b9bc26a